hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncClusteringService.java

```diff
@@ -103,6 +103,12 @@ protected Pair<CompletableFuture, ExecutorService> startService() {
    * Update the write client to be used for clustering.
    */
   public synchronized void updateWriteClient(BaseHoodieWriteClient writeClient) {
-    this.clusteringClient.updateWriteClient(writeClient);
+    this.clusteringClient.updateWriteClient(writeClient.createNewClient());
```

Comment:
If we also schedules a plan in the async compaction/clustering service, then this fix might be necessary, but let's also keep caution to the double memory occupation for the fs view on the driver here.

Is it possible we move the scheduling into the writer itself and the service is only responsible for execution?

Reply:
Checked these two services, and the plan was scheduled ourside and enqueued through `HoodieAsyncService#enqueuePendingAsyncServiceInstant`, not sure why there are fs view in consistency issues here? Because the plan execution does not really access the fs view, just read the source files and write it out (with create handle).

Reply:
yeah